Classes/Issues/Merge/IssueMergeSectionController.swift

```diff
@@ -65,13 +65,25 @@ MergeButtonDelegate {
         var viewModels = [ListDiffable]()
 
         if object.contexts.count > 0 {
-            let success = object.contexts.reduce(true, { $0 && $1.state == .success })
-            viewModels.append(IssueMergeSummaryModel(
-                title: success ?
-                    NSLocalizedString("All checks passed", comment: "")
-                    : NSLocalizedString("Some checks failed", comment: ""),
-                state: success ? .success : .failure
-            ))
+            let states = object.contexts.map { $0.state }
+            let state: IssueMergeSummaryModel.State
+            let stateDescription: String
+            let containsFailure = states.contains(.failure) || states.contains(.error)
+            switch states {
+            case _ where containsFailure:
+                state = .failure
+                stateDescription = NSLocalizedString("Some checks failed", comment: "")
+            case let states where states.contains(.pending) && !containsFailure:
+                state = .pending
+                stateDescription = NSLocalizedString("Merge status pending", comment: "")
+            case let states where states.reduce(true, { $0 && $1 == .success }):
```

Comment:
This line blew my mind for just a few seconds, took a bit to work out what was happening, could it be made simpler at all do you think? (Obviously just a personal thing)

Reply:
It hasn't changed from [what it was before](https://github.com/rnystrom/GitHawk/pull/1533/files#diff-7c180300a62c299e2c10544bd45d9223L68), but I do agree its readability could be improved. We can add a `allSucceeded` bool at the top, just like [`containsFailure`](https://github.com/rnystrom/GitHawk/pull/1533/files#diff-7c180300a62c299e2c10544bd45d9223R71) (at a super tiny performance penalty for always executing that `reduce`)?

Reply:
And / or we (re)pitch [this proposal](https://github.com/apple/swift-evolution/pull/664) that adds `all` to `Sequence`? 😀

Reply:
Fair point about it not changing, not saying you have to change it though!

I wonder if something like `states.contains(where: { $0 != .success })` or similar (I think that's a thing...)

Or yea as you say, move a var to the top!

Also that all syntax looks really interesting 🤔 Would be super helpful actually

Reply:
> I wonder if something like `states.contains(where: { $0 != .success })` or similar (I think that's a thing...)

Yeah but then you would need [`!states.contains(where: { $0 != .success })`](https://github.com/apple/swift-evolution/pull/664/files#diff-b5c7d543f27e9ba8db174753956310eeR21) (notice the `!` in front), which is even less readable (unless you don't know `reduce`, I suppose).

Reply:
Super fair comment, I'll leave it to you to pick the solution with the least devils - if that means keeping as is I'm fine with that!